File: app/src/main/java/com/legue/axel/lolsummonertool/database/viewmodel/ItemStatViewModel.java
package com.legue.axel.lolsummonertool.database.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.annotation.NonNull;

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase;
import com.legue.axel.lolsummonertool.database.model.item.ItemStat;

public class ItemStatViewModel extends AndroidViewModel {

    private SummonerToolDatabase database;

    public ItemStatViewModel(@NonNull Application application) {
        super(application);
        database = SummonerToolDatabase.getInstance(this.getApplication());
    }

    public LiveData<ItemStat> getItemStatByItemId(int itemId) {
        return database.itemStatDao().getItemStatByItemId(itemId);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/viewmodel/ItemStatViewModel.kt
package com.legue.axel.lolsummonertool.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase
import com.legue.axel.lolsummonertool.database.model.item.ItemStat

class ItemStatViewModel(application: Application) : AndroidViewModel(application) {

    private val database: SummonerToolDatabase = SummonerToolDatabase.getInstance(this.getApplication())

    fun getItemStatByItemId(itemId: Int): LiveData<ItemStat> {
        return database.itemStatDao().getItemStatByItemId(itemId)
    }
}

